README.md

```diff
@@ -1,23 +1,176 @@
 
 ## News
+
 2017.02.08: ModelDB publicly available! Try it out and contribute.
 
-## Setup
+## Setup and Installation
+
+Watch a video of the setup and installation process [here](https://youtu.be/rmNnG3-bd6s).
+
+1. **Clone the repo**
+
+    ```git
+    git clone https://github.com/mitdbg/modeldb
+    ```
+
+2. **Install dependencies**
+
+    ModelDB requires Linux or MacOS. The code below shows how you can install the dependencies on each of them. A detailed list of all the dependencies with the recommended and required versions can be found [here](docs/required_software.md).
+    
+    Depending on the client you're using, we assume you have the following already installed:
+    - scikit-learn client:
+        - Python 2.7\*\*
+        - [pip](https://pip.pypa.io/en/stable/installing/)
+        - [scikit-learn](http://scikit-learn.org/stable/install.html) 0.17\*\*
+    - spark.ml client:
+        - Java 1.8+
+        - Spark 2.0.0\*\*
+
+    **(\*\*) = Must have exact version**
+
+    On OSX, we also assume that you have [homebrew](https://brew.sh/) installed.
+
+    **On Mac OSX:**
+
+    ```bash
+    # Use homebrew to install the dependencies
+    brew install sqlite
+    brew install maven
+    brew install node
+    brew install sbt # for spark.ml client only
+    brew install mongodb
+
+    # ModelDB works only with Thrift 0.9.3 and 0.10.0. If you do not have thrift installed, install via brew.
+
+    brew install thrift
+    ```
+
+    **On Linux:**
+
+    ```bash
+    apt-get update
+    sudo apt-get install sqlite
+    sudo apt-get install maven
+    sudo apt-get install sbt # for spark.ml client only
+    sudo apt-get install nodejs # may need to symlink node to nodejs. "cd /usr/bin; ln nodejs node"
+    sudo apt-get install -y mongodb-org # further instructions here: https://docs.mongodb.com/manual/tutorial/install-mongodb-on-ubuntu/
+
+    # install thrift. [path_to_thrift] is the installation directory
+    cd [path_to_thrift]
+    wget http://mirror.cc.columbia.edu/pub/software/apache/thrift/0.9.3/thrift-0.9.3.tar.gz
+    tar -xvzf thrift-0.9.3.tar.gz
+    cd thrift-0.9.3
+    ./configure
+    make
+    export PATH=[path_to_thrift]/:$PATH
+    ```
+
+    For Linux, you can also refer to [this script](https://github.com/mitdbg/modeldb/blob/master/docs/install_on_linux.sh). 
+
+3. **Build**
+    
+    ModelDB is composed of three components: the ModelDB **server**, the ModelDB **client libraries**, and the ModelDB **frontend**.
+
+    In the following, **[path_to_modeldb]** refers to the directory into which you have cloned the modeldb repo and **[thrift_version]** is 0.9.3 or 0.10.0 depending on your thrift version (check by running ```thrift -version```).
+
+    ```bash
+    # run the script to set up the sqlite and the mongodb databases that modeldb will use
+    # note that this clears any previous modeldb databases
+    # this also starts mongodb
+    cd [path_to_modeldb]/server/codegen
+    ./gen_sqlite.sh
```

Comment:
Add a BOLD warning that this should be done only once. that this will clean out any data in the DB